datafusion/core/src/physical_planner.rs

```diff
@@ -1792,11 +1792,19 @@ impl DefaultPhysicalPlanner {
                             Err(e) => return Err(e),
                         }
                     }
-                    Err(e) => stringified_plans
-                        .push(StringifiedPlan::new(InitialPhysicalPlan, e.to_string())),
+                    Err(err) => {
+                        return Ok(Some(Arc::new(ExplainExec::new(
+                            SchemaRef::new(Schema::new(vec![arrow_schema::Field::new(
```

Comment:
I think this change means that the explain will not have the logical plan, but instead will have only the error message for creating the physical plan as the "Final Logical Plan"